File: jsweet-client/src/main/java/json_schema/events/BotDeathEvent.java
package json_schema.events;

public class BotDeathEvent extends Event {

	public static final String TYPE = "botDeathEvent";

	public BotDeathEvent() {
		super(TYPE);
	}

	public Integer getVictimId() {
		return (Integer) $get("victimId");
	}
}

File: bot-api/python/src/robocode_tank_royale/bot_api/events/bot_death_event.py
from dataclasses import dataclass

from .bot_event import BotEvent


@dataclass(frozen=True, repr=True)
class BotDeathEvent(BotEvent):
    """
    Event occurring when another bot has died.

    Attributes:
        victim_id (int): The ID of the bot that has died.
    """

    victim_id: int

    def __post_init__(self) -> None:
        # validate victim_id
        if self.victim_id < 0:
            raise ValueError(f"victim_id must be a non-negative integer, got {self.victim_id}.")
